pkg/nmi/server/server.go

```diff
@@ -380,3 +386,29 @@ func handleTermination() {
 	log.Infof("Exiting with %v", exitCode)
 	os.Exit(exitCode)
 }
+
+func listPodIDsWithRetry(kubeClient k8s.Client, logger *log.Entry, podns, podname, rqClientID string, maxAttempts int) (*[]aadpodid.AzureIdentity, error) {
+	attempt := 0
+	var err error
+	var podIDs *[]aadpodid.AzureIdentity
+
+	for {
+		podIDs, err = kubeClient.ListPodIds(podns, podname)
+		if err == nil && len(*podIDs) != 0 {
+			if len(rqClientID) == 0 {
```

Comment:
there are two kinds of error, one is that after a few seconds, we still did not find the AzureAssignedIdentity and another is that it never moved to Assigned. Not able to gather that different from retry attempts and varying time durations for them..?

Reply:
The retry attempts are not different based on the states. The final desired state is `Assigned`. So I think this encompasses all possible scenarios.